### PR TITLE
LPS-165263 | Add Autom. test FrontendJSCustomDialog#CanRenderOpenAlertModal

### PR DESCRIPTION
**Pull request by deborasoliveira**
**Test Plan**

- Given: Enabled custom Dialog
- And When: run JS code: Liferay.Util.openAlertModal({message: "Test Alert Modal", onConfirm: ()=>{}})
- Then: text "Test Alert Modal" is present in the modal

**JIRA Ticket**
https://issues.liferay.com/browse/LPS-165263

### CODE REVIEW

File: modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FrontendJSCustomDialog.testcase [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+			PortalInstances.tearDownCP();
+		}
+		else {
+			SystemSettings.configureSystemSetting(
+				enableSetting = "false",
+				settingFieldName = "Enabled");
+		}
+	}
+
+	@description = "LPS-165262"
```
Comment (john-co) [inline modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FrontendJSCustomDialog.testcase:44]: Add more text to the description:
`This test validates that the liferay custom alert modal dialog can render.`

File: modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FrontendJSCustomDialog.testcase [hunk 2]
```diff
@@ -0,0 +1,58 @@
+@component-name = "portal-frontend-infrastructure"
+definition {
+
+	property custom.properties = "upgrade.database.auto.run=false";
+	property osgi.modules.includes = "frontend-data-set-sample-web";
```
Comment (john-co) [inline modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FrontendJSCustomDialog.testcase:5]: Remove these two lines. Not needed

File: modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FrontendJSCustomDialog.testcase [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+	property portal.acceptance = "true";
+	property portal.release = "true";
+	property portal.upstream = "true";
+	property testray.component.names = "Frontend Dataset";
+	property testray.main.component.name = "Frontend Dataset";
```
Comment (john-co) [inline modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FrontendJSCustomDialog.testcase:10]: replace with `User Interface` or create a new `testray.component.names`

Comment (john-co) [reply modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FrontendJSCustomDialog.testcase:10]: main component name = Frontend JS API
component name = Frontend Custom Dialog

**Comment by deborasoliveira** [comment]
ci:test:sf

**Comment by deborasoliveira** [comment]
ci:test:frontend:nocompile

File: modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FrontendJSCustomDialog.testcase (hunk 3 above)
Comment (deborasoliveira) [reply modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FrontendJSCustomDialog.testcase]: done!

File: modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FrontendJSCustomDialog.testcase (hunk 2 above)
Comment (deborasoliveira) [reply modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FrontendJSCustomDialog.testcase]: done!

File: modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FrontendJSCustomDialog.testcase (hunk 1 above)
Comment (deborasoliveira) [reply modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FrontendJSCustomDialog.testcase]: done!

**Comment by eltonccdantas** [comment]
Hi @deborasoliveira, I analyzed your request:


**Not Unique Fails:**

The fails below also happened on [EE Development Acceptance (master)](https://testray.liferay.com/home/-/testray/case_results?testrayBuildId=2041279412): 


- [LocalFile.ManagementToolbar#OrderButtonAsAToggleCanDisplayTooltip](https://testray.liferay.com/home/-/testray/case_results/2041716769?redirect=https%3A%2F%2Ftestray.liferay.com%2Fhome%2F-%2Ftestray%2Fcase_results%2Findex%3FtestrayBuildId%3D2041543717%26testrayRunId%3D2041715974%26cur%3D1%26delta%3D200%26orderByCol%3Dstatus_sortable%26orderByType%3Dasc)

**Unique fails:**

[LocalFile.RemoteApps#HasMandatoryFieldErrorMessagesWhenEmpty](https://testray.liferay.com/home/-/testray/case_results/2041716103?redirect=https%3A%2F%2Ftestray.liferay.com%2Fhome%2F-%2Ftestray%2Fcase_results%2Findex%3FtestrayBuildId%3D2041543717%26testrayRunId%3D2041715974%26cur%3D1%26delta%3D200%26orderByCol%3Dstatus_sortable%26orderByType%3Dasc) 

[LocalFile.WYSIWYGUsecase#CanDisplayFormattedHTMLFromCopyPaste](https://testray.liferay.com/home/-/testray/case_results/2041717125?redirect=https%3A%2F%2Ftestray.liferay.com%2Fhome%2F-%2Ftestray%2Fcase_results%2Findex%3FtestrayBuildId%3D2041543717%26testrayRunId%3D2041715974%26cur%3D1%26delta%3D200%26orderByCol%3Dstatus_sortable%26orderByType%3Dasc)


[LocalFile.RightToLeftInfrastructure#RTLDirectionStyleAppliedToTextField](https://testray.liferay.com/home/-/testray/case_results/2041716884?redirect=https%3A%2F%2Ftestray.liferay.com%2Fhome%2F-%2Ftestray%2Fcase_results%2Findex%3FtestrayBuildId%3D2041543717%26testrayRunId%3D2041715974%26cur%3D1%26delta%3D200%26orderByCol%3Dstatus_sortable%26orderByType%3Dasc)

[Top Level Build](https://testray.liferay.com/home/-/testray/case_results/2041716072?redirect=https%3A%2F%2Ftestray.liferay.com%2Fhome%2F-%2Ftestray%2Fcase_results%2Findex%3FtestrayBuildId%3D2041543717%26testrayRunId%3D2041715974%26cur%3D1%26delta%3D200%26orderByCol%3Dstatus_sortable%26orderByType%3Dasc)

**Comment by eltonccdantas** [comment]
Hey Debora, it seems to me your request is ok, there's only one fail and not related to your PR as you can see on this [comparison](https://testray.liferay.com/home/-/testray/runs/compare?compare=true&view=details&statusA=2&testrayRunIdB=2041715974&testrayRunIdA=2041279452&statusB=3)

**Comment by eltonccdantas** [comment]
ci:foward

**Comment by john-co** [comment]
ci:forward

**Pull request closed without merging**